components/multipart-stream-builder.rst

```diff
@@ -0,0 +1,63 @@
+Multipart Stream Builder
+========================
+
+A multipart stream is a special kind of stream that is used to transfer files over HTTP. There is currently no PSR-7 support for multipart streams as they are considered to be normal streams with a special content. A multipart stream HTTP request may look like this:
+
+.. code-block:: none
+
+    POST / HTTP/1.1
+    Host: example.com
+    Content-Type: multipart/form-data; boundary=578de3b0e3c46
+
+    --578de3b0e3c46
+    Content-Disposition: form-data; name="foo"
+    Content-Length: 15
+
+    A normal stream
+    --578de3b0e3c46
+    Content-Disposition: form-data; name="bar"; filename="bar.png"
+    Content-Length: 71
+    Content-Type: image/png
+
+    ?PNG
+    
+    ???
+    IHDR??? ??? ?????? ???? IDATxc???51?)?:??????IEND?B`?
```

Comment:
The question mars are `HDR          �� �   IDATxc���` which does not make any sense either.. I just represent binary data..